static/app/views/settings/projectSourceMaps/associations.tsx

```diff
@@ -0,0 +1,122 @@
+import {Link} from 'react-router';
+import styled from '@emotion/styled';
+
+import ClippedBox, {ClipFade} from 'sentry/components/clippedBox';
+import {CopyToClipboardButton} from 'sentry/components/copyToClipboardButton';
+import {Hovercard} from 'sentry/components/hovercard';
+import Placeholder from 'sentry/components/placeholder';
+import TextOverflow from 'sentry/components/textOverflow';
+import {t, tct, tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {DebugIdBundleAssociation} from 'sentry/types/sourceMaps';
+import useOrganization from 'sentry/utils/useOrganization';
+
+function AssociationsBody({associations}: {associations: DebugIdBundleAssociation[]}) {
+  const organization = useOrganization();
+
+  return (
+    <ClippedBoxWithoutPadding
+      clipHeight={210}
+      btnText={t('+ %s more', associations.length - associations.slice(0, 4).length)}
+      buttonProps={{
+        priority: 'default',
+        borderless: true,
+      }}
+    >
+      <NumericList>
+        {associations.map(({release, dist}) => (
+          <li key={release}>
+            <ReleaseContent>
+              <ReleaseLink
+                to={`/organizations/${organization.slug}/releases/${release}/`}
+              >
+                <TextOverflow>{release}</TextOverflow>
+              </ReleaseLink>
+              <CopyToClipboardButton
+                text={release}
+                borderless
+                size="zero"
+                iconSize="sm"
+              />
+            </ReleaseContent>
+            {!dist?.length ? (
+              <NoAssociations>
+                {t('No dists associated with this release')}
+              </NoAssociations>
+            ) : (
+              tct('Dist: [dist]', {dist: dist.join(', ')})
+            )}
+          </li>
+        ))}
+      </NumericList>
+    </ClippedBoxWithoutPadding>
+  );
+}
+
+type Props = {
+  associations?: DebugIdBundleAssociation[];
```

Comment:
i think this could be not optional since you check for the array before rendering this component

Reply:
so this won't be at some point, but since we are doing some sort of transition I added the optional type for now